### PR TITLE
623 custom ingress tls

### PR DESCRIPTION
**Pull request by mtgoncalves1**
## Overview
Closes https://gitlab.com/architect-io/architect-cli/-/issues/623.  
Add TLS to ingress spec to allow customer to specify their own certs.

## Changes
Define ingress TLS spec and ingress TLS config

## Test
Try any string value for crt, key, and/or ca.
```
name: ingress-tls
services:
  api:
    interfaces:
      main:
        ingress:
          subdomain: api
          tls:
            crt: ${{ secrets.api_crt }}
            key: ${{ secrets.api_key }}
            ca: ${{ secrets.api_ca }}
```
$ architect register    

## Docs
If this is a new feature or changes functionality, provide a link to the ticket for documentation additions or updates.

### CODE REVIEW

**Review by tjhiggins — APPROVED** [review]
LGTM - I'll let Michael merge

**Comment by mtgoncalves1** [comment]
> LGTM - I'll let Michael merge

Thank you!

**Comment by mueschm** [comment]
Looks good, but will wait for API PR to merge in.

**Review by TylerAldrich — APPROVED** [review]

**Pull request merged**
